.travis.yml

```diff
@@ -9,7 +9,38 @@ cache:
   directories:
     - $HOME/.m2
 
-script: ./mvnw clean verify
+env:
+  matrix:
+    - SPRING_VERSION=4.1.0.RELEASE
```

Comment:
> Is it really necessary to build with all these versions

I did it today and a lot of them introduced issues.

> doesn't that slow down testing a lot

Travis should execute them in parallel.

> would expect the patch releases to keep the api compatible.

API yes (mostly, at least) but the implementation sometimes changes in an imcompatible way.